### PR TITLE
Fix assorted frontend regressions

### PR DESCRIPTION
**Pull request by mboynes**
Recent updates introduced some regressions to the FE build, as well as the general UI and UX of the plugin in non-gutenberg screens. This PR aims to resolve those issues.

### CODE REVIEW

File: client/src/components/byline-profiles/index.js [hunk 1]
```diff
@@ -99,12 +99,10 @@ const BylineProfiles = ({
   const removeItem = (id) => {
     const index = profiles.findIndex((item) => item.id === id);
     if (0 <= index) {
-      this.setState({
```
Comment (ostowe) [inline client/src/components/byline-profiles/index.js:102]: I honestly don't know how this was ever working...probably it wasn't

File: client/src/components/byline-profiles/index.js [hunk 2]
```diff
@@ -118,7 +116,7 @@ const BylineProfiles = ({
         const newSearchResults = rawResults.filter(
           (result) => 0 > currentIds.indexOf(result.id),
         );
-        setSearchResults({ newSearchResults });
+        setSearchResults(newSearchResults);
```
Comment (ostowe) [inline client/src/components/byline-profiles/index.js:119]: This also seemed to cause a fatal error, at least on classic editor posts

File: client/src/init-user-link-metabox.js [hunk 3]
```diff
@@ -22,11 +22,11 @@ const initUserLinkMetaBox = () => {
     <UserLinkMetaBox
       linkUserPlaceholder={linkUserPlaceholder}
       linkedToLabel={linkedToLabel}
-      postId={postId}
+      postId={+ postId}
```
Comment (ostowe) [inline client/src/init-user-link-metabox.js:25]: does this coerce it to a number? 😮

File: client/src/components/byline-profiles/index.js (hunk 1 above)
Comment (mboynes) [reply client/src/components/byline-profiles/index.js:102]: I noticed when bisecting that it was recently converted from a `React.Component` class to a function, so probably only broken as of recently

File: client/src/init-user-link-metabox.js (hunk 3 above)
Comment (mboynes) [reply client/src/init-user-link-metabox.js:25]: yep! A little trick Pattie taught me

File: client/src/components/byline-profiles/index.js (hunk 1 above)
Comment (ostowe) [reply client/src/components/byline-profiles/index.js:102]: yeah, figured that must be the case

**Review by daronspence — APPROVED** [review]
🤘  all makes sense to me. Not sure if the `+ number-ishStringVar` thing is worth saving 8 chars over but 

¯\_(ツ)_/¯

**Review by kaitlinbolling — APPROVED** [review]
🎉  Nice fixes/enhancements. I checked this out locally to compare against production and it looks & works great.

**Pull request merged**